pkg/configurator/client.go

```diff
@@ -21,44 +17,9 @@ import (
 )
 
 // NewConfigurator implements configurator.Configurator and creates the Kubernetes client to manage namespaces.
-func NewConfigurator(configClient configClientset.Interface, stop <-chan struct{}, osmNamespace, meshConfigName string,
-	msgBroker *messaging.Broker) (Configurator, error) {
-	return newConfigurator(configClient, stop, osmNamespace, meshConfigName, msgBroker)
-}
-
-func newConfigurator(configClient configClientset.Interface, stop <-chan struct{}, osmNamespace string, meshConfigName string,
-	msgBroker *messaging.Broker) (*client, error) {
-	listOption := configInformers.WithTweakListOptions(func(opt *metav1.ListOptions) {
-		opt.FieldSelector = fields.OneTermEqualSelector(metav1.ObjectNameField, meshConfigName).String()
-	})
-
-	meshConfigInformerFactory := configInformers.NewSharedInformerFactoryWithOptions(
-		configClient,
-		k8s.DefaultKubeEventResyncInterval,
-		configInformers.WithNamespace(osmNamespace),
-		listOption,
-	)
-
-	// informerFactory without listOptions
-	configInformerFactory := configInformers.NewSharedInformerFactoryWithOptions(
-		configClient,
-		k8s.DefaultKubeEventResyncInterval,
-		configInformers.WithNamespace(osmNamespace),
-	)
-
-	informerCollection := informerCollection{
-		meshConfig:          meshConfigInformerFactory.Config().V1alpha2().MeshConfigs().Informer(),
-		meshRootCertificate: configInformerFactory.Config().V1alpha2().MeshRootCertificates().Informer(),
-	}
-
-	cacheCollection := cacheCollection{
-		meshConfig:          informerCollection.meshConfig.GetStore(),
-		meshRootCertificate: informerCollection.meshRootCertificate.GetStore(),
-	}
-
+func NewConfigurator(informerCollection *informers.InformerCollection, osmNamespace, meshConfigName string, msgBroker *messaging.Broker) (*client, error) {
```

Comment:
@steeling @shashankram  What are your opinions on this pattern? I've seen it used before, but the linter doesn't like it. I don't have strong opinions either way; just curious how y'all feel

Reply:
I think we should likely export it, all the fields are unexported so it still has to be intiialized through this mechanism